### ISSUE TITLE
LibreNMS API as source broken

### ISSUE THREAD

**Issue by illgetmine16** [ytti/oxidized#2913]
I tried to move to the latest (oxidized/oxidized:latest) and then I get an error about the LibreNMS API returning no usable nodes. It's sending back code 200, and curl gets the same results as before. I'm assuming in newer releases Oxidized is looking for a different format or something? I had to roll it back to 0.29.1-11-g3d9f986 in order for it to work again.

**Comment by jacobw** [comment]
I'm getting the same issue. I thought it was me but I was getting correct results via curl so I couldn't figure it out.

**Comment by jacobw** [comment]
Same issue here, rolling back to that version fixes it for me too.

**Comment by bkeifer** [comment]
Also seeing this issue.  It appears that 82e434b is the breaking change.

**Comment by jacobw** [comment]
Perhaps this is related:

```
**Disclaimer**: currently only tested with netbox as the source
```

**Comment by bkeifer** [comment]
I also added it in a comment on the commit itself, but below is a sample host list returned by the LibreNMS API:

```
[
    {
        "hostname": "host1.example.com",
        "os": "iosxe",
        "ip": "172.20.129.99",
        "group": "default"
    },
    {
        "hostname": "host2.example.com",
        "os": "aosw8",
        "ip": "172.20.251.39",
        "group": "default"
    }
]
```

**Comment by krejler** [comment]
For me it's also working in 0.29.1-15-gdafa835 but all version after is broken.

**Comment by varesa** [comment]
Looking at the [commit before the one that's identified as breaking](https://github.com/ytti/oxidized/blob/41ebd79a42002bb2e4093e62653e59d8bd330a54/lib/oxidized/source/http.rb), paying attention to the `hosts_location` config parameter:

```
def load(node_want = nil)
      nodes = []
      ### Get raw response
      data = JSON.parse(read_http(node_want))
      
      ### Traverse to hosts_location, IF defined
      data = string_navigate(data, @cfg.hosts_location) if @cfg.hosts_location?
      
      ### 
      data.each do |node|
        ### omited transforms, nodes gets appended to
        nodes << keys
      end
      nodes
    end
```

Compared to the [following commit](https://github.com/ytti/oxidized/blob/82e434b180c583a6cb735806394435165257d128/lib/oxidized/source/http.rb):
```
    def check_pagination(response, http, headers, node_want)
      node_data = []
      if @cfg.pagination?
        ### ignoring the pagination case for now
      # since new feature; dont break curent configs
      else
        ### get raw response
        data = JSON.parse(response.body)
        ### only append to node_data IF hosts_location is set
        node_data += string_navigate(data, @cfg.hosts_location) if @cfg.hosts_location?
      end
      node_data
    end
```

So it seems that the case where `hosts_location` is not set and the devices are at the root of the JSON object returned was lost and `check_pagination` will only return an empty array?

**Comment by varesa** [comment]
testing a fix in https://github.com/cybercom-finland/oxidized/commit/9b00130ef4412fb45aaf9900006f64984892eee6

**Comment by varesa** [comment]
I was about to hit submit on a PR for the above...

but turns out it is already fixed by f9aa9556e8b1d35e352baba8a5a48771d40029d3 in the meanwhile

**Comment by jacobw** [comment]
Dupe of [2897](https://github.com/ytti/oxidized/issues/2897)

**Comment by robertcheramy** [comment]
Duplicate of #2897 and fixed in f9aa955.
Thanks @varesa an @jacobw for spoting it.
